scratchwork/fsdp_min/fsdp.py

```diff
@@ -86,11 +108,15 @@ def save_checkpoint(
         ):
             torch.save(model_state_dict, path.joinpath("model.bin"))
             torch.save(optim_state_dict, path.joinpath("optim.bin"))
+            # Scaler state is automatically the same across ranks.
+            scaler_state_dict = scaler.state_dict()
+            torch.save(scaler_state_dict, path.joinpath("scaler.bin"))
```

Comment:
The behavior looks to vary over torch versions; in the image tested, it doesn't cause an issue.

In the most recent `main` this should work when `use_amp` is the same between runs, but would raise an unnecessary exception when switching from `use_amp=False` to `use_amp=True` with the same checkpoint.  I'll add an enabled on save and enabled + file-exists on load check to avoid version issues.